### PR TITLE
feat(docdb): add copyTagsToSnapshot property to the DatabaseCluster Construct

### LINKED ISSUE #30090: (docdb): copyTagsToSnapshot is missing in the DatabaseCluster Construct
### Describe the feature

In the [DatabaseCluster](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_docdb.DatabaseCluster.html) Construct, [copyTagsToSnapshot](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_docdb.CfnDBCluster.html#copytagstosnapshot) is missing.

### Use Case

Use this when you want to copy tags to snapshots.

### Proposed Solution

Add copyTagsToSnapshot  to the DatabaseCluster Construct.

```ts
const cluster = new docdb.DatabaseCluster(this, 'Database', {
  masterUser: {
    username: 'myuser', // NOTE: 'admin' is reserved by DocumentDB
    excludeCharacters: '\"@/:', // optional, defaults to the set "\"@/" and is also used for eventually created rotations
    secretName: '/myapp/mydocdb/masteruser', // optional, if you prefer to specify the secret name
  },
  instanceType: ec2.InstanceType.of(ec2.InstanceClass.MEMORY5, ec2.InstanceSize.LARGE),
  vpc,
  copyTagsToSnapshot: true, // Add
});
```

### Other Information

_No response_

### Acknowledgements

- [X] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### CDK version used

2.140.0

### Environment details (OS name and version, etc.)

MacOS

### PR DESCRIPTION
**Pull request by mazyu36**
### Issue # (if applicable)

Closes #30090 

### Reason for this change
As described in the issue.

### Description of changes
Add copyTagsToSnapshot property to the DatabaseCluster Construct.

### Description of how you validated changes
Add both unit tests and integ tests.

### Checklist
- [x] My code adheres to the [CONTRIBUTING GUIDE](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md) and [DESIGN GUIDELINES](https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md)

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/aws-cdk-lib/aws-docdb/lib/cluster.ts [hunk 1]
```diff
@@ -217,6 +217,13 @@ export interface DatabaseClusterProps {
    * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-attribute-deletionpolicy.html
    */
   readonly securityGroupRemovalPolicy?: RemovalPolicy;
+
+  /**
+   * Whether to copy tags to the snapshot when a snapshot is created.
+   *
+   * @default - not copy tags to the snapshot
```
Comment (aaythapa) [inline packages/aws-cdk-lib/aws-docdb/lib/cluster.ts:224]: nit: `@default - false`

File: packages/aws-cdk-lib/aws-docdb/test/cluster.test.ts [hunk 2]
```diff
@@ -1114,6 +1114,71 @@ describe('DatabaseCluster', () => {
...
+      instanceType: ec2.InstanceType.of(ec2.InstanceClass.BURSTABLE2, ec2.InstanceSize.SMALL),
+      vpc,
+      copyTagsToSnapshot: true,
+    });
+
+    // THEN
+    Template.fromStack(stack).hasResourceProperties('AWS::DocDB::DBCluster', {
+      CopyTagsToSnapshot: true,
+    });
+  });
```
Comment (aaythapa) [inline packages/aws-cdk-lib/aws-docdb/test/cluster.test.ts:1181]: nit: can be combined into one test

```suggestion
    test.each([false, true])('cluster with copyTagsToSnapshot set', (value) => {
        const stack = testStack();
        const vpc = new ec2.Vpc(stack, 'VPC');
      
        // WHEN
        new DatabaseCluster(stack, 'Database', {
          masterUser: {
            username: 'admin',
            password: cdk.SecretValue.unsafePlainText('tooshort'),
          },
          instanceType: ec2.InstanceType.of(ec2.InstanceClass.BURSTABLE2, ec2.InstanceSize.SMALL),
          vpc,
          copyTagsToSnapshot: value,
        });
      
        // THEN
        Template.fromStack(stack).hasResourceProperties('AWS::DocDB::DBCluster', {
          CopyTagsToSnapshot: value,
        });
    }
```

**Review by aaythapa — COMMENTED** [review]
Thanks for making this change! Some tiny nits

File: packages/aws-cdk-lib/aws-docdb/lib/cluster.ts (hunk 1 above)
Comment (mazyu36) [reply packages/aws-cdk-lib/aws-docdb/lib/cluster.ts]: @aaythapa 
I fixed it.

File: packages/aws-cdk-lib/aws-docdb/test/cluster.test.ts (hunk 2 above)
Comment (mazyu36) [reply packages/aws-cdk-lib/aws-docdb/test/cluster.test.ts]: @aaythapa 
Thank you for your suggestion.
I applied it.

**Review by aaythapa — APPROVED** [review]
🚀

**Pull request merged**